pkg/service/active_directory_membership_service.go

```diff
@@ -74,7 +74,7 @@ func (svc *ActiveDirectoryMembershipService) GetActiveDirectoryMembershipById(id
 func (svc *ActiveDirectoryMembershipService) GetActiveDirectoryMembershipByName(name string) (*nimbleos.ActiveDirectoryMembership, error) {
 	params := &param.GetParams{
 		Filter: &param.SearchFilter{
-			FieldName: nimbleos.VolumeFields.Name,
+			FieldName: param.NewString(nimbleos.VolumeFields.Name),
```

Comment:
It doesn't seem right that we are using VolumeFields in all these places that are unrelated to volumes.  It should be &nimbleos.ActiveDirectoryMembershipFields.Name, right?

Reply:
Hmm I'm not sure; it was already there when I branched from SM's default branch. (I pasted a snippet below from `ServicesTemplate.ftl`). Are you sure we need to change it?
```
// Get${DataObjectName}ByName - method returns a pointer "${DataObjectName}"
func (svc *${DataObjectName}Service) Get${DataObjectName}ByName(name string) (*nimbleos.${DataObjectName}, error) {
    params := &param.GetParams{
        Filter: &param.SearchFilter{
            FieldName: param.NewString(nimbleos.VolumeFields.Name),
            Operator:  param.EQUALS.String(),
            Value:     name,
        },
    }
    ${DataObjectName?uncap_first}Resp, err := svc.objectSet.GetObjectListFromParams(params)
    if err != nil {
        return nil, err
    }

    if len(${DataObjectName?uncap_first}Resp) == 0 {
        return nil, nil
    }

    return ${DataObjectName?uncap_first}Resp[0],nil
    </#if>
}
```

Reply:
I understand this issue was not introduced by your PR.  Am I sure we need to change it?  No, it presumably works fine the way it is.  But it seems wrong philosophically to filter a GET request on a field from a different object type.  I think we should change VolumeFields.Name to ${DataObjectName}Fields.Name.  If that change is found to cause a problem somewhere, it will be interesting to understand why.